packages/provider/provider.0.0.9/opam

```diff
@@ -0,0 +1,40 @@
+opam-version: "2.0"
+synopsis: "Dynamic Dispatch with Traits"
+maintainer: ["Mathieu Barbin"]
+authors: ["Mathieu Barbin"]
+license: "ISC"
+homepage: "https://github.com/mbarbin/provider"
+doc: "https://mbarbin.github.io/provider/"
+bug-reports: "https://github.com/mbarbin/provider/issues"
+depends: [
+  "dune" {>= "3.16"}
+  "ocaml" {>= "5.2"}
+  "bisect_ppx" {dev & >= "2.8.3"}
+  "ppx_js_style" {dev & >= "v0.17"}
```

Comment:
```suggestion
  "bisect_ppx" {with-dev-setup & >= "2.8.3"}
  "ppx_js_style" {with-dev-setup & >= "v0.17"}
```
Are you sure this is `dev` and not [`with-dev-setup`](https://opam.ocaml.org/doc/Manual.html#pkgvar-with-dev-setup)? The latter is the flag to install "developer tools"

Reply:
Afaik the `with-dev-setup` flag was introduced for linting and instrumentation, for the people that wanted to add a dependency to ocamlformat or ocp-indent for example. The history for this is here: https://github.com/ocaml/opam/issues/4959

The CI will not install these and run the linting steps, in the same way as it does not do it with dev (`dev` is only true if a user wants to install the head of the master branch of a opam package with `opam install --dev ...` and is never true for releases).

> Given that I also have a test package in the same repo, is it important that I add these with-dev-setup section to the main package, or is that enough to simply list them as dependencies to the test package?

I am not sure what you mean. They can stay also in the main package, it depends on how you run the tests and the lint. To use them in your CI is enough to make sure that opam is called with the `--with-dev-setup` flag

Reply:
Thanks a lot @mseri ! 

I updated the repo to use `--with-dev-setup` and added `ocamlformat` there as well. https://github.com/mbarbin/provider/pull/10

Given that there are no other changes, I also reported the changes to the opam file directly in this PR. Is that OK, even if technically the opam file from the 0.0.9 archive will not be exactly equal to the opam file checked into the opam repo for that version? (or is it cleaner to cancel this release and create a new `0.0.10` release?).

I took the opportunity to apply the same change to other repos, as well as a [template](https://github.com/mbarbin/opam-package-template) I use. Thank you for your help!

Reply:
> I am not sure what you mean. They can stay also in the main package, it depends on how you run the tests and the lint. To use them in your CI is enough to make sure that opam is called with the --with-dev-setup flag

What I mean is this: In the provider repo, I have two opam files.
- [provider.opam](https://github.com/mbarbin/provider/blob/main/provider.opam)
- [provider-tests.opam](https://github.com/mbarbin/provider/blob/main/provider-tests.opam)
The CI for that project now includes [--with-dev-setup](https://github.com/mbarbin/provider/blob/3a46d62cb848066d98b403915179747e74cbfe06/.github/workflows/ci.yml#L34)

I am guessing that when you are at the root of the repo and run:
```sh
$ opam install . --with-doc --with-test --with-dev-setup
```
it is enough for the dev dependencies to be listed in any of the present opam files to be picked up by the install command. Thus, my question is : what are the pros of including the dev-setup dependencies of `provider.opam`. Is there a context in which not having them there is going to make something fail?

Reply:
Yes that will install the dev-setup and test dependencies for all the packages. If you only need it for development and not for the main package, then you can just have them all in the `-tests` package